packages/botonic-api/src/rest/routes/events.ts

```diff
@@ -1,38 +1,177 @@
-      },
-    }
-    return res.json(event)
+  .get(checkSchema({ eventId: eventIdParamSchema }), async (req, res) => {
+    try {
+      const errors = validationResult(req)
+      if (!errors.isEmpty()) {
+        res.status(400).send({ errors: errors.array() })
+        return
+      }
+
+      const params = matchedData(req, { locations: ['params'] })
+      const dp = dataProviderFactory(process.env.DATA_PROVIDER_URL)
+      const event = await dp.getEvent(params.eventId)
+      if (!event) {
+        res
+          .status(404)
+          .send({ error: `Event with ID '${params.eventId}' not found` })
+        return
+      }
+      res.status(200).send(event)
+    } catch (e) {
+      res.status(500).send({ error: e.message })
+    }
   })
-  .put((req, res) => {
-    return res.send(`PUT HTTP method on event/${req.params.eventId} resource`)
+  .put(checkSchema({ eventId: eventIdParamSchema }), async (req, res) => {
+    const errors = await validateBotonicEventData(req)
+    if (!errors.isEmpty()) {
+      res.status(400).send({ errors: errors.array({ onlyFirstError: true }) })
+      return
+    }
+
+    try {
+      const errors = validationResult(req)
+      if (!errors.isEmpty()) {
+        res.status(400).send({ errors: errors.array() })
+        return
+      }
+
+      const params = matchedData(req, { locations: ['params'] })
+      const updatedEvent = matchedData(req, {
+        locations: ['body'],
+      }) as BotonicEvent
+
+      const dp = dataProviderFactory(process.env.DATA_PROVIDER_URL)
+      const event = await dp.getEvent(params.eventId)
+      if (!event) {
+        res
+          .status(404)
+          .send({ error: `Event with ID '${params.eventId}' not found` })
+        return
+      }
+      await dp.updateEvent(updatedEvent)
+      res.status(200).send(updatedEvent)
+    } catch (e) {
+      res.status(500).send({ error: e.message })
+    }
   })
-  .patch((req, res) => {
-    return res.send(`PATCH HTTP method on event/${req.params.eventId} resource`)
+  .patch(checkSchema({ eventId: eventIdParamSchema }), async (req, res) => {
+    const errors = await validateBotonicEventData(req, true)
+    if (!errors.isEmpty()) {
+      res.status(400).send({ errors: errors.array({ onlyFirstError: true }) })
+      return
+    }
+
+    try {
+      const errors = validationResult(req)
+      if (!errors.isEmpty()) {
+        res.status(400).send({ errors: errors.array() })
+        return
+      }
+
+      const params = matchedData(req, { locations: ['params'] })
+      const newEventData = matchedData(req, {
+        locations: ['body'],
+      }) as Partial<BotonicEvent>
+
+      const dp = dataProviderFactory(process.env.DATA_PROVIDER_URL)
+      const event = await dp.getEvent(params.eventId)
+      if (!event) {
+        res
+          .status(404)
+          .send({ error: `Event with ID '${params.eventId}' not found` })
+        return
+      }
+      const updatedEvent = Object.assign(event, newEventData)
```

Comment:
Extra: `const updatedEvent = {...event, ...newEventData}` (I think it's more readable this way and Object.assign modifies the first param which can cause undesired side effects)

Reply:
Yes, I did it this way in the first place but I got a TS error that made me add an `as BotonicEvent` to make sure `updateEvent` has the right type and since `event` can be modified (without making any side effects) I decided to use `Object.assign`.

But I agree with you that it's more readable with deconstruction, changed!